### DISCUSSION TITLE
Cannot preserve openness while adding / removing nodes from tree?

### DISCUSSION THREAD

**Discussion by chungwu** [Lodin/react-vtree#50] [Q&A]
From the new 3.x API, it looks like if the nodes in the tree changes (by adding / removing nodes), a new treeWalker is required, which will reset all state, including the existing openness of individual nodes.  Is that right?  Is there a way to preserve openness while mutating the tree?

**Comment by Lodin** [comment]
Yes, there is an [`async`](https://github.com/Lodin/react-vtree#async-boolean) prop that does the trick. It allows preserving the tree state between tree updates.
